seed/analysis_pipelines/eui.py

```diff
@@ -59,73 +62,95 @@ def _get_valid_meters(property_view_ids):
-                    meter_readings = []
-                    continue
-
-                # readings are already normalized to kBtu on import so we only need the reading value
-                meter_readings.append(meter_reading.reading)
-                previous_month = meter_reading.start_time.month
-                streak = streak + 1
-
-                # just use the most recent 12 months found
-                if streak >= 12:
-                    break
-
-            # just use the first meter found
-            if streak >= 12:
-                break
-
-        if streak < 12:
-            invalid_meter_2.append(property_view.id)
-            continue
-        meter_readings.reverse()
-        meter_readings_by_property_view[property_view.id] = meter_readings
+        # get all readings that started AND ended between end_time and a year prior
+        property_meter_readings = [
+            SimpleMeterReading(reading.start_time, reading.end_time, reading.reading)
+            for reading in MeterReading.objects.filter(
+                meter__property=property_view.property,
+                meter__type__in=VALID_METERS,
+                end_time__lte=end_time,
+                start_time__gte=end_time - TIME_PERIOD
+            ).order_by('start_time')
+        ]
+
+        meter_readings_by_property_view[property_view.id] = property_meter_readings
 
     errors_by_property_view_id = {}
     for pid in invalid_area:
         if pid not in errors_by_property_view_id:
             errors_by_property_view_id[pid] = []
         errors_by_property_view_id[pid].append(EUI_ANALYSIS_MESSAGES[ERROR_INVALID_GROSS_FLOOR_AREA])
-    for pid in invalid_meter_1:
-        if pid not in errors_by_property_view_id:
-            errors_by_property_view_id[pid] = []
-        errors_by_property_view_id[pid].append(EUI_ANALYSIS_MESSAGES[ERROR_INSUFFICIENT_METER_READINGS])
-    for pid in invalid_meter_2:
+    for pid in invalid_meter:
         if pid not in errors_by_property_view_id:
             errors_by_property_view_id[pid] = []
         errors_by_property_view_id[pid].append(EUI_ANALYSIS_MESSAGES[ERROR_INVALID_METER_READINGS])
 
     return meter_readings_by_property_view, errors_by_property_view_id
 
 
+def _get_days_in_reading(meter_reading):
+    """Returns a list of datetime.datetime days that the reading covers/touches
+
+    :param meter_reading: List[SimpleMeterReading | MeterReading]
+    :return: List[datetime.datetime], days (at midnight, timezone unaware)
+    """
+    start = datetime.datetime(
+        meter_reading.start_time.year,
+        meter_reading.start_time.month,
+        meter_reading.start_time.day,
+    )
+    end = datetime.datetime(
+        meter_reading.end_time.year,
+        meter_reading.end_time.month,
+        meter_reading.end_time.day,
+    )
+
+    all_days = []
+    current_day = start
+    while current_day <= end:
+        all_days.append(current_day)
+        current_day += relativedelta.relativedelta(days=1)
```

Comment:
This appears to be a pretty inefficient algorithm, but should be fine for now. Technically constant time since all meter readings must fit within a year (365 days).